gateway-js/src/index.ts

```diff
@@ -568,6 +569,9 @@ export class ApolloGateway implements GatewayInterface {
     this.queryPlanStore.clear();
     this.apiSchema = supergraph.apiSchema();
     this.schema = addExtensions(this.apiSchema.toGraphQLJSSchema());
+
+    assert(!this.config.queryPlannerConfig?.typeConditionedFetching, "Type conditions are not supported in the gateway");
```

Comment:
We might as well do this in the constructor if this is just a configuration option. In the `issueConfigurationWarningsIfApplicable` method seems like an appropriate place for this concern, possibly with a rename -> `validateConfigAndEmitWarnings` or something like that?